# oteodoro changes:
# Patch status:  testing
# Changed from Partial RELRO to Full RELRO
# For use in the hardened profile

License header of ELF/Driver.cpp:

//===- Driver.cpp ---------------------------------------------------------===//
//
// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
// See https://llvm.org/LICENSE.txt for license information.
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
//
//===----------------------------------------------------------------------===//

diff '--color=auto' -urp lld.orig/ELF/Driver.cpp lld/ELF/Driver.cpp
--- lld.orig/ELF/Driver.cpp	2025-11-03 01:35:47.000000000 -0800
+++ lld/ELF/Driver.cpp	2025-11-10 06:26:10.599319083 -0800
@@ -1633,10 +1633,10 @@ static void readConfigs(Ctx &ctx, opt::I
   ctx.arg.zNodefaultlib = hasZOption(args, "nodefaultlib");
   ctx.arg.zNodelete = hasZOption(args, "nodelete");
   ctx.arg.zNodlopen = hasZOption(args, "nodlopen");
-  ctx.arg.zNow = getZFlag(args, "now", "lazy", false);
+  ctx.arg.zNow = getZFlag(args, "now", "lazy", true);          // Changed by oteodoro to enable Full RELRO by default
   ctx.arg.zOrigin = hasZOption(args, "origin");
   ctx.arg.zPacPlt = getZFlag(args, "pac-plt", "nopac-plt", false);
-  ctx.arg.zRelro = getZFlag(args, "relro", "norelro", true);
+  ctx.arg.zRelro = getZFlag(args, "relro", "norelro", false);  // Changed by oteodoro to enable Full RELRO by default
   ctx.arg.zRetpolineplt = hasZOption(args, "retpolineplt");
   ctx.arg.zRodynamic = hasZOption(args, "rodynamic");
   ctx.arg.zSeparate = getZSeparate(args);
Only in lld/ELF: Driver.cpp.orig
Only in lld/ELF: Driver.cpp.rej
